### DISCUSSION TITLE
Duration with a start date?

### DISCUSSION THREAD

**Discussion by nazarewk** [jotaen/klog#226] [Feature ideas & discussions]
Sometimes it is relevant to tell when a period started and it's duration (I noticed periods with same summary get merged to 1, so i don't know how many `#off=break` i took throughout the day.

I propose following syntax:

```
2022-11-08
#work=Employer1
    8:30 - 17:00  standard workday
    12:00-30m #workout
    13:00-15m walking the dog
    23:00+2h #emergency 
```

after writing it down I noticed it might be (visually) hard to discern it from regular Range entries

This could work too, should be easier to discern:

```
2022-11-08
#work=Employer1
    8:30 - 17:00  standard workday
    -30m@12:00 #workout
    -15m@13:00 walking the dog
    +2h@23:00 #emergency 
```

Somewhat related to as it solves similar issues:
- https://github.com/jotaen/klog/discussions/46
- https://github.com/jotaen/klog/discussions/109

**Comment by jotaen** [comment]
Thanks for sharing your ideas! I agree that the first example would look quite similar to range entries. Regarding the second one, I’m wondering what the functional difference would be to writing:

```
2022-11-08
#work=Employer1
    8:30 - 17:00  standard workday
    -30m @12:00 #workout
    -15m @13:00 walking the dog
    +2h @23:00 #emergency 
```

I.e., just putting the point in time into the summary text? That way you could capture that info, even though the notation would be “your own convention”, rather than a predefined syntax. I think a dedicated syntax would only make sense if there was a functional difference, so if that would allow klog to interpret or evaluate the data differently. But maybe I haven’t fully understood your use-case yet, so please correct me if I’m wrong here.

> I noticed periods with same summary get merged to 1, so i don't know how many `#off=break` i took throughout the day.

Which command are you referring to here?

Regarding `klog tags`, I think it would be a cool and useful idea to also display the count info: https://github.com/jotaen/klog/issues/227

**Reply by nazarewk** [reply]
I'm pretty sure `klog pause` adds on top of the previous pause with same summary, yesterday I made 3+, but today i'm seeing just one for aggregated time

**Reply by nazarewk** [reply]
as for the value added to pauses:
- there could be accurate information (start, stop, duration) for all entries, not just ranges
- you could track how often you take breaks

alternative approach during pause giving same functionality could be:
1. split currently active range (open ended or not)
    1. stop right now
    2. insert `0m <summary>` entry
    3. start a new range with same summary and same end date
2. until pause is finished:
	1. move currently active (new) range's start to `now`
	2. remove the entry when going over it's end date and stop tracking

**Reply by jotaen** [reply]
Ah, thanks for clarifying. Yes, `klog pause` indeed does add to an existing negative duration, in case its summary text is the same. (Otherwise, if the summary text is different, it would append a new entry.) Back then, when I added the `klog pause` command, I thought that would be useful behaviour, but I actually realize that it might be a bit confusing. So I think this could be improved: https://github.com/jotaen/klog/issues/228. With the auto-extending behaviour being optional, it would be possible to decide whether you want to extend or append.

I hope that would help already, but I’m otherwise also open to change/extend the behaviour of `klog pause` further.
